src/race-factory.ts

```diff
@@ -0,0 +1,22 @@
+import { delay } from './delay';
+
+export function raceFactory<T>(
+	timeout: number,
+	callback: (...args: any[]) => Promise<any>,
+): (...args: any[]) => Promise<T> {
+	return async (...args) => {
+		let finished = false;
+		try {
+			return await Promise.race([
+				callback(...args),
+				delay(timeout).then(() => {
```

Comment:
I know that you are just using it, but to share some knowledge, you can use the setTimeout of 'timers/promises'; it is a lib native in node.js and is a promise, but I think for compatibility with older versions of node.js we can live like this :smile: